operator/src/main.rs

```diff
@@ -168,20 +167,64 @@ async fn install_trustee_configuration(client: Client, namespace: String) -> Res
         Err(e) => error!("Failed to create the KBS configuration: {e}"),
     }
 
-    // TODO replace this creation with a per-machine one.
-    // This secret's address is `default/machine/root`.
-    match trustee::generate_secret(
+    // Create a dummy secret. If there is no secret created early, trustee-operator does not allow
+    // for updating them later as it changes the KbsConfig structure to include no new keys.
+    let clevis_ctx = operator::ClevisContextData {
+        client,
+        trustee_namespace: cocl.spec.trustee.namespace.clone(),
+        kbs_config: cocl.spec.trustee.kbs_config_name.clone(),
+    };
+    match trustee::generate_secret(clevis_ctx, "dummy").await {
+        Ok(_) => info!("Generate dummy secret"),
+        Err(e) => error!("Failed to create dummy secret: {e}"),
+    }
+
+    Ok(())
+}
+
+async fn install_register_server(client: Client) -> Result<()> {
+    let cocl = list_confidential_clusters(client.clone(), client.default_namespace()).await?;
+
+    match register_server::create_register_server_rbac(client.clone()).await {
+        Ok(_) => info!("Register server RBAC created/updated successfully"),
+        Err(e) => error!("Failed to create register server RBAC: {e}"),
+    }
+
+    let meta = cocl.metadata;
+    let owner_name = meta.name.context("ConfidentialCluster had no name")?;
+    let owner_uid = meta.uid.context("ConfidentialCluster had no UID")?;
+    let owner_reference = OwnerReference {
+        api_version: ConfidentialCluster::api_version(&()).to_string(),
+        kind: ConfidentialCluster::kind(&()).to_string(),
+        name: owner_name.to_string(),
+        uid: owner_uid.to_string(),
+        ..Default::default()
```

Comment:
Please, set explicitly `block_owner_deletion` to true in this way, k8s blocks the cocl cr deletion until the deployment is gone. Also `controller` should be set to true.